### PR TITLE
MD Fixes for Model/ApiConnector/Client.php

### PR DESCRIPTION
**Pull request by simon-letch**
…/app/code/Dotdigitalgroup/Email/Model/Apiconnector/Client.php:5	The class Client has 54 public methods and attributes. Consider reducing the number of public items to less than 45.

/opt/bamboo/bamboo-agent-home/xml-data/build-dir/M2-L4938-SE/magento2/app/code/Dotdigitalgroup/Email/Model/Apiconnector/Client.php:5	The class Client has an overall complexity of 111 which is very high. The configured complexity threshold is 100.

I've removed api methods that we weren't using to drop the public items down to 44. I see no point in culling them all at this point as they may be required in the future.

### CODE REVIEW

File: Model/Apiconnector/Client.php [hunk 1]
```diff
@@ -270,55 +270,6 @@ public function deleteAddressBookContact($addressBookId, $contactId)
...
-    }
-
-    /**
-     * Gets a contact by email address.
-     *
-     * @param string $email
-     *
-     * @return mixed
-     */
-    public function getContactByEmail($email)
```
Comment (adeelq) [inline Model/Apiconnector/Client.php]: @simon-letch We using this in project in following classes
/Dotdigitgroup/Email/Model/Sync/Contact/Update.php
/Dotdigitgroup/Email/Block/Customer/Account/Books.php

File: Model/Apiconnector/Client.php [hunk 2]
```diff
@@ -340,30 +291,6 @@ public function getAddressBooks()
...
 
     /**
-     * Gets an address book by ID.
-     *
-     * @param $id
-     *
-     * @return null
-     * @throws \Exception
-     */
-    public function getAddressBookById($id)
```
Comment (adeelq) [inline Model/Apiconnector/Client.php]: @simon-letch We using this in project in following classes
/Dotdigitgroup/Email/Block/Customer/Account/Books.php

File: Model/Apiconnector/Client.php [hunk 3]
```diff
@@ -483,32 +410,6 @@ public function getDataFields()
...
 
     /**
-     * Updates a contact.
-     *
-     * @param string|int $contactId
-     * @param $data
-     *
-     * @return object
-     */
-    public function updateContact($contactId, $data)
```
Comment (adeelq) [inline Model/Apiconnector/Client.php]: @simon-letch this is used everywhere in project

File: Model/Apiconnector/Client.php [hunk 4]
```diff
@@ -861,107 +741,6 @@ public function getAccountInfo()
...
-    }
-
-    /**
-     * Resubscribes a previously unsubscribed contact.
-     *
-     * @param $apiContact
-     *
-     * @throws \Exception
-     */
-    public function postContactsResubscribe($apiContact)
```
Comment (adeelq) [inline Model/Apiconnector/Client.php]: @simon-letch We using this in project in following classes
/Dotdigitgroup/Email/Model/Sync/Contact/Update.php

**Comment by simon-letch** [comment]
I'm going to abandon this. I've unchecked the tasks in trello.

**Pull request closed without merging**